main.nf

```diff
@@ -1,130 +1,575 @@
+    str += bcfrac ? "--bcfrac=${bcfrac} " : ""
+    str += nopreflight ? "--nopreflight " : ""
+    str += no_accept_extreme_coverage ? "" : "--accept-extreme-coverage "
     return str
 }
 
-//###### Start
-log.info "     \\|/"
-log.info "  ----*----   N e u t r o n S t a r (${version})"
-log.info "     /|\\" 
-log.info "  supernova run --id=${p.id} --fastqs=${p.fastqs} ${supernova_optional()} "
-log.info "  outdir = ${p.outdir}"
+def samples = []
+if (params.samples == null) { //We don't have sample JSON/YAML file, just use cmd-line
+    assert params.id : "Missing --id option"
+    assert params.fastqs : "Missing --fastqs option"
+    s = []
+    s += params.id
+    s += params.fastqs
+    s += TenX_optional(params.sample, params.lanes, params.indices, params.project)
+    s += supernova_optional(params.maxreads, params.bcfrac, params.nopreflight, params.no_accept_extreme_coverage)
+    samples << s
+}
+
+params.samples = []
+
+for (sample in params.samples) {
+    assert sample.id : "Error in input parameter file"
+    assert sample.fastqs : "Error in input parameter file"
+    s = []
+    s += sample.id
+    s += sample.fastqs
+    s += TenX_optional(sample.sample, sample.lanes, sample.indices, sample.project)
+    s += supernova_optional(sample.maxreads, sample.bcfrac, sample.nopreflight, sample.no_accept_extreme_coverage)
+    samples << s
+}
+
+//Extra parameter evaluation
+for (sample in params.samples) {
+    assert sample.id.size() == sample.id.findAll(/[A-z0-9\\.\-]/).size() : "Illegal character(s) in sample ID: ${sample.id}."
+    for(folder in sample.fastqs.split(",")) {
+      def f = new File(folder)
+      assert f.exists() : "Non-existent folder specified for --fastqs: ${folder}"
+    }
+}
+
+if (!(workflow.runName ==~ /[a-z]+_[a-z]+/)) {
+  custom_runName = workflow.runName
+}
+
+if ( workflow.profile == 'awsbatch') {
+  // AWSBatch sanity checking
+  if (!params.awsqueue || !params.awsregion) exit 1, "Specify correct --awsqueue and --awsregion parameters on AWSBatch!"
+  // Check outdir paths to be S3 buckets if running on AWSBatch
+  // related: https://github.com/nextflow-io/nextflow/issues/813
+  if (!params.outdir.startsWith('s3:')) exit 1, "Outdir not on S3 - specify S3 Bucket to run on AWSBatch!"
+  // Prevent trace files to be stored on S3 since S3 does not support rolling files.
+  if (workflow.tracedir.startsWith('s3:')) exit 1, "Specify a local tracedir or run without trace! S3 cannot be used for tracefiles."
+}
+
+// Stage config files
+ch_multiqc_config = file(params.multiqc_config, checkIfExists: true)
+ch_output_docs = file("$baseDir/docs/output.md", checkIfExists: true)
+
+// Header log info
+log.info nfcoreHeader()
+def summary = [:]
+if (workflow.revision) summary['Pipeline Release'] = workflow.revision
+summary['Run Name']         = custom_runName ?: workflow.runName
+// TODO nf-core: Report custom parameters here
+summary['Reads']            = params.reads
```

Comment:
I don't think that this is used in your pipeline? And `params.samples` isn't reported here.. Might be worth going through these...

Reply:
Ah yes. I used to print this in previous versions of the pipeline